aten/src/ATen/native/mkl/SpectralOps.cpp

```diff
@@ -46,104 +50,134 @@ namespace at { namespace native {
-  // (basically data_slices[i:i+num]), and keeps track of the slices it reads
-  // data from, i.e., from_slice_data, using from_slice_indices, a vector
-  // containing the index of the from_slice_data slice.
-
-  // Compute the indices for the first from_slice_data
-  std::vector<int64_t> from_slice_indices(signal_ndim);  // up to before last signal dim
-  int64_t remainder = i;
-  // set last signal dim values
-  int64_t from_slice_offset = 0;
-  for (int64_t d = signal_ndim - 1; d >= 0; d--) {
-    int64_t dim_size = output.size(d);
-    int64_t dim_idx = remainder % dim_size;
-    remainder = remainder / dim_size;
-    from_slice_indices[d] = dim_idx;
-    if (d == 0) {
-      from_slice_offset += dim_idx * output.stride(d);
-    } else if (dim_idx != 0) {
-      from_slice_offset += (dim_size - dim_idx) * output.stride(d);
-    }
-  }
+static __ubsan_ignore_undefined__  // UBSAN gives false positives on using negative indexes with a pointer
+void _fft_fill_with_conjugate_symmetry_slice(
+    Range range, at::ArrayRef<bool> is_mirrored_dim, IntArrayRef signal_half_sizes,
+    IntArrayRef in_strides, const scalar_t * in_ptr,
+    IntArrayRef out_strides, scalar_t * out_ptr) {
+  const int64_t ndim = signal_half_sizes.size();
+  DimVector iter_index(ndim, 0);
 
-  // First to_slice_data and from_slice_data
-  scalar_t *to_slice_data = data + i * size_last_dim * 2;
-  scalar_t *from_slice_data = data + from_slice_offset;
-
-  while (num > 0) {
-    // Fill to_slice_data from values in from_slice_data
-    for (int64_t j = start_last_dim_idx; j < size_last_dim; j++) {
-      // multiply index by 2 because of the last complex dim has size 2
-      int64_t to_idx = j * 2;
-      int64_t from_idx = (size_last_dim - j) * 2;
-      to_slice_data[to_idx] = from_slice_data[from_idx];
-      to_slice_data[to_idx + 1] = -from_slice_data[from_idx + 1];
-    }
-    // Compute the next to_slice_data and from_slice_data slices
-    to_slice_data += size_last_dim * 2;
-    for (int64_t d = signal_ndim - 1; d >= 0; d--) {
-      // Compute the next index at this dimension using conjugate symmetry
-      // Break out of this loop if nothing carries over
-      from_slice_indices[d] = (from_slice_indices[d] + 1) % output.size(d);
-      if (d > 0) {
-        // At d > 0 nonbatch dim, to get next from_slice_data offset
-        //   1. if this dim idx becomes 1, will need to add (size - 1) * stride
-        //   2. otherwise, will need to subtract stride
-        if (from_slice_indices[d] == 0) {
-          // Subtract. Carries over to previous dimension
-          from_slice_data -= output.stride(d);
-        } else if (from_slice_indices[d] == 1) {
-          // Dimension index becomes 1
-          // Doesn't carry over to previous dimension
-          from_slice_data += (output.size(d) - 1) * output.stride(d);
-          break;
+  auto advance_index = [&] {
+    for (int64_t i = 1; i < iter_index.size(); ++i) {
```

Comment:
auto i here too to match the type of .size()

Reply:
Changed to `size_t`.